### PR TITLE
Remove `UnsafeBeyond2Pow53Minus1` type suffixes

### PR DESCRIPTION
**Pull request by lorisleiva**
Both RPC and RPC Subscriptions packages now use custom JSON serialization to safely convert large integers as `bigints` from the server requests/responses. This means we are no longer handling `bigints` that are unsafe below `2^53-1` which also means we can remove the `UnsafeBeyond2Pow53Minus1` suffixes to all types currently using it.

Note that the `F64UnsafeSeeDocumentation` type is excluded from this refactoring as the bigint serialization only affects integers.

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3456?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3456** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3456?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3455** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3455?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3452** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3452?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3451** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3451?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3450** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3450?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3449** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3449?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3448** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3448?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3443** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3443?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lorisleiva and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/rpc-types/src/typed-numbers.ts [hunk 1]
```diff
@@ -1,24 +1,12 @@
-/**
- * FIXME(solana-labs/solana/issues/30341) Beware that any value above 9007199254740991 may be
- * truncated or rounded because of a downcast to JavaScript `number` between your calling code and
- * the JSON-RPC transport.
- */
-export type U64UnsafeBeyond2Pow53Minus1 = bigint;
+export type U64 = bigint;
+export type I64 = bigint;
```
Comment (buffalojoec) [inline packages/rpc-types/src/typed-numbers.ts:2]: I don't have a strong opinion here, but do we need these aliases?

**Review by buffalojoec — APPROVED** [review]
beautiful

File: packages/rpc-types/src/typed-numbers.ts (hunk 1 above)
Comment (lorisleiva) [reply packages/rpc-types/src/typed-numbers.ts:2]: I also don't have strong opinions here. I guess it's nice to mark that a certain bigint is using 8 bytes max and is signed/unsigned. Happy to add another PR if we decide to dissolve them though.

Comment (mcintyre94) [reply packages/rpc-types/src/typed-numbers.ts:2]: I'd lean toward removing them too, but also not a strong opinion. We're not using them for the relevant codecs and I don't think we should, which makes me think they're probably not adding much.

**Review by mcintyre94 — APPROVED** [review]
So good!
Added via Giphy

**Review by steveluscher — APPROVED** [review]
Added via Giphy

File: packages/rpc-types/src/typed-numbers.ts (hunk 1 above)
Comment (steveluscher) [reply packages/rpc-types/src/typed-numbers.ts:2]: There's marking to be helpful, but on the flipside there's trusting that the type enforces the runtime behaviour, which here it does not (ie. these could, in fact, be negative and TypeScript would not complain).

Comment (lorisleiva) [reply packages/rpc-types/src/typed-numbers.ts:2]: Sounds like we're all leaning toward dissolving them. Follow up PR incoming.

**Comment by lorisleiva** [comment]
### Merge activity

* **Oct 27, 6:03 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3456).

File: packages/rpc-types/src/typed-numbers.ts (hunk 1 above)
Comment (lorisleiva) [reply packages/rpc-types/src/typed-numbers.ts:2]: https://github.com/solana-labs/solana-web3.js/pull/3467

**Pull request merged**